storage/notification_cleaner_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package storage_test
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+
+	"github.com/Peripli/service-manager/storage"
+
+	"github.com/Peripli/service-manager/storage/storagefakes"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Notification cleaner", func() {
+	var (
+		ctx           context.Context
+		cancel        context.CancelFunc
+		wg            *sync.WaitGroup
+		fakeStorage   *storagefakes.FakeStorage
+		nc            *storage.NotificationCleaner
+		expectedError = errors.New("*Expected*")
+	)
+
+	BeforeEach(func() {
+		fakeStorage = &storagefakes.FakeStorage{}
+		ctx, cancel = context.WithCancel(context.Background())
+		wg = &sync.WaitGroup{}
+		nc = &storage.NotificationCleaner{
+			Storage:  fakeStorage,
+			Settings: storage.DefaultSettings(),
+		}
+	})
+
+	AfterEach(func() {
+		if ctx.Err() == nil {
+			cancel()
+		}
+		wg.Wait()
```

Comment:
Not needed

Reply:
This is for safety. Some tests in the future might require this.